scripts/property_utils.sh

```diff
@@ -36,19 +30,24 @@ function set_property() {
   VALUE=$2
   FILE=$3
 
-  REGEXP="${1//\./\\\.}\s*="
-  if [ $(grep $REGEXP $FILE | wc -l) -eq 0 ]; then
-    write_prop $1 $2 $3
-  else
-    # delete line of specified property
+  REGEXP="${PROPERTY//\./\\.}\\s*="
+
+  if grep -q "$REGEXP" "$FILE"; then
+     # delete line of specified property
     LINE_COUNT=$(cnt_lines $FILE)
-    sed -i $SED_DISABLE_BACKUP "/${REGEXP}/d" $FILE
+
+    # Create new file and replace afterwards. Avoids corruption if update fails
+    sed -e /${REGEXP}/d "$FILE" > "${FILE}.new" && mv "${FILE}.new" "${FILE}"
```

Comment:
Yes -e flag is for Mac. 
Having the SED_DISABLE_BACKUP variable caused to create a backup file with quotes in the end (filename.properties"). Another option is to inline OS check. That would actually look nicer since it doesn't create a global variable as well.